src/components/CreateNewProjectWizard/Gatsby/SelectStarterList.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+import React, { PureComponent } from 'react';
+import styled from 'styled-components';
+import Scrollbars from 'react-custom-scrollbars';
+import { Tooltip } from 'react-tippy';
+
+import Divider from '../../Divider';
+import ExternalLink from '../../ExternalLink';
+import CodesandboxLogo from '../../CodesandboxLogo';
+import StrokeButton from '../../Button/StrokeButton';
+
+import { COLORS } from '../../../constants';
+
+type Props = {
+  updateStarter: (string, boolean) => void,
+  handleShowMore: () => void,
+  selectedStarter: string,
+  isVisible: boolean,
+  starters: Array<any>,
+  paginationIndex: number,
+  lastIndex: number,
+};
+
+/*
+Gatsby starter selection dialog.
+A starter object from starter.yml contains the following data:
+[
+   {
+      "url": "https://wonism.github.io/",
+      "repo": "https://github.com/wonism/gatsby-advanced-blog",
+      "description": "n/a",
+      "tags": [
+        "Portfolio",
+        "Redux"
+      ],
+      "features": [
+        "feature items"
+      ]
+  }, ... 
+]
+*/
+class SelectStarterList extends PureComponent<Props> {
+  node: any;
+
+  handleUpdateStarter = (starter: string, removeSelection: boolean) => {
+    const { updateStarter } = this.props;
+
+    updateStarter(starter, removeSelection);
+
+    // Scroll to top as the selected starter will be the first entry
+    this.node.scrollTop(0);
+  };
+
+  prepareUrlForCodesandbox(repoUrl: string) {
+    // Remove http protocol
+    const sandboxUrl = `https://codesandbox.io/s/${repoUrl.replace(
+      /(^\w+:|^)\/\//,
+      ''
+    )}`;
+    // Remove .com from github.com --> to have /s/github/repo
+    return sandboxUrl.replace(/\.com/, '');
+  }
+
+  render() {
+    const {
+      isVisible,
+      starters,
+      selectedStarter,
+      paginationIndex,
+      lastIndex,
+      handleShowMore,
+    } = this.props;
+
+    return (
+      <MainContent isVisible={isVisible}>
+        <ScrollContainer innerRef={node => (this.node = node)}>
+          <StarterList>
+            {starters.map((starter, index) => (
+              <StarterItem key={index}>
+                <StarterItemTitle>
+                  <StarterItemHeading
+                    selected={selectedStarter === starter.repo}
+                    onClick={() =>
+                      this.handleUpdateStarter(

```

Comment:
There's a lot of passing down/abstraction of the `updateFieldValue` function (from `CreateNewProjectWizard`) to ultimately end up here.

The chain right now seems to be:

1. `CreateNewProjectWizard` passes it to `MainPane` as `updateFieldValue` prop
2. `MainPane` passes it to `ProjectStarter` as `onSelect` prop
3. `ProjectStarter` passes it to `SelectStarterList` as `updateStarter` prop
4. `SelectStarterList` abstracts it into function `handleUpdateStarter`

This was a bit hard to follow when I was in `ProjectStarter` and wanted to see where `handleOnSelect` was getting its parameters set.

I wonder if there's a way to simplify this or at least keep the naming consistent when passing it around?